src/include/execution_common.hpp

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#ifndef MSCCLPP_EXECUTION_COMMON_HPP_
+#define MSCCLPP_EXECUTION_COMMON_HPP_
+
+#include <mscclpp/proxy_channel.hpp>
+#include <mscclpp/sm_channel.hpp>
+
+namespace mscclpp {
+
+constexpr int MAX_CHANNEL = 16;
+constexpr int MAX_CHANNEL_PER_OPERATION = 8;
+constexpr int MAX_OPERATION = 64;
```

Comment:
Can we make these configurable?

Reply:
I think we'd better to make this config static. It used by the structure `Operation`. And the kernel code uses this structure to execute the generated plan. Static structure would be more efficient for execution.

Reply:
Thank you, let's discuss this in the next version.